src/components/select-field/BaseSelectField.js

```diff
@@ -1,16 +1,18 @@
 // @flow
 import * as React from 'react';
 import classNames from 'classnames';
-import scrollIntoViewIfNeeded from 'scroll-into-view-if-needed';
 import uniqueId from 'lodash/uniqueId';
 
+import { scrollItemIntoView } from '../../utils/dom';
 import IconCheck from '../../icons/general/IconCheck';
 import SelectButton from '../select-button';
 import DatalistItem from '../datalist-item';
 import type { SelectOptionValueProp, SelectOptionProp } from './props';
 
 import './SelectField.scss';
 
+const OVERLAY_WRAPPER_CLASS = 'overlay-wrapper';
```

Comment:
What if the field does not reside within an Overlay component? Does scrollItemIntoView behave correctly? What if the BaseSelectField resides within the HTML body itself? Shouldn't the body be scrolled?

Reply:
Yes, I think we should be passing in the class that denotes what should be scrolled. Instead of hardcoding overlay-wrapper as the class that should be scrolled, we need to define on a case-by-case basis which class should be scrolled and pass that in to BaseSelectField / SelectorDropdown

Reply:
I would name the variable starting with CLASS_OVERLAY_WRAPPER.
Also search the codebase to see if there are other uses of this class name, and if so, move the constant to src/constants